database/agents.go

```diff
@@ -248,7 +248,8 @@ func (db *DB) ActiveAgentsByTarget(target string) (agents []mig.Agent, err error
 	}
 	rows, err := txn.Query(fmt.Sprintf(`SELECT DISTINCT ON (queueloc) id, name, queueloc,
 		version, pid, starttime, destructiontime, heartbeattime, refreshtime, status,
-		mode, environment, tags
+		mode, environment, tags,
+		(SELECT loadername FROM loaders WHERE loaders.queueloc = agents.queueloc LIMIT 1)
```

Comment:
This is only executed when scheduling new actions, so I think it's fine if it isn't the fastest query in the world. Still,you seem to be retrieving data for the solve purpose of logging, and I can't help but wonder if that shouldn't just be put in a separate query that's part of the logging function `logAgentAction`.